### PR TITLE
Fix schema to handle error

### LINKED ISSUE #59: schema.check_write_permission can throw exceptions
This chunk of code...

https://github.com/uclahs-cds/pipeline-Nextflow-config/blob/c211cc381a5a011c7c00ea89a260ec9d0ba98252/config/schema/schema.config#L41-L43

... does not account for the fact that `getParentFile()` can return `null` ([docs](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#getParentFile--)).

In the specific case that a bare relative path is provided to this method, an unhandled (and supremely unhelpful) exception will be raised:

```console
$ NXF_WORK=./test/work nextflow run ./main.nf -c /hot/code/nwiltsie/pipelines/pipeline-align-DNA/test/global.config -c ./test/a_mini_n2-picard.config --output_dir a_mini_n2-picard
...
ERROR - Unable to parse config file: '/hot/code/nwiltsie/pipelines/pipeline-align-DNA/./test/a_mini_n2-picard.config'

  Cannot invoke method exists() on null object
```

### PR DESCRIPTION
**Pull request by zhuchcn**
1. In `schema.check_path`, if the directory is relative, and the directory doesn't exist, it will try to call `getParentFile` until it becomes `null`  and failed with the error of `no exists() method for null object`. So I turned the path to an absolute path.
2. Also added a try catch block to print what parameter that causes an error, which can save a lot of time for debugging.

- [x] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).

- [x] I have reviewed the [Nextflow pipeline standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3193890/Nextflow+pipeline+standardization).

- [x] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using \[AD_username (or 5 letters of AD if AD is too long)]-\[brief_description_of_branch].

- [x] I have set up or verified the branch protection rule following the [github standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3190380/GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

- [ ] I have added my name to the contributors listings in the
``metadata.yaml`` and the ``manifest`` block in the `nextflow.config` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

- [ ] I have added the changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

- [ ] I have updated the version number in the `metadata.yaml` and `manifest` block of the `nextflow.config` file following [semver](https://semver.org/), or the version number has already been updated. (*Leave it unchecked if you are unsure about new version number and discuss it with the infrastructure team in this PR.*)

- [ ] I have tested the config being added or modified. Outline the tests below.

Closes #59

### CODE REVIEW

**Comment by nwiltsie** [comment]
Nice! This should fix #59.

File: config/schema/schema.config [hunk 1]
```diff
@@ -56,6 +56,7 @@ schema {
     */
     check_path = { String p, String mode ->
         def File file = new File(p)
+        file = new File(file.getAbsolutePath())
```
Comment (nwiltsie) [inline config/schema/schema.config:59]:
```suggestion
        file = file.getAbsoluteFile()
```
From [the docs](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#getAbsoluteFile--):
> Returns the absolute form of this abstract pathname. Equivalent to `new File(this.getAbsolutePath()`

**Review by nwiltsie — APPROVED** [review]
Looks good! I've added a suggestion, but it's not required.

**Review by yashpatel6 — APPROVED** [review]
Looks good! Can you also update the CHANGELOG to indicate the fix?

File: config/schema/schema.config (hunk 1 above)
Comment (zhuchcn) [reply config/schema/schema.config:59]: Ah that's better. `.getAbsolutePath` returns a string.

**Comment by zhuchcn** [comment]
Change log updated!

**Pull request merged**